contrib/duy-nguyen/srfi/159/internal/write.scm

```diff
@@ -152,7 +152,7 @@
           (let lp ((ls ls) (res '()))
             (cond
              ((null? ls)
-              (cons 1 res))
+              (append-reverse res '(1)))
```

Comment:
LGTM, this is already fixed upstream almost the same: `(append (reverse res) '(1))`